### PR TITLE
tests: net: socket: offload_dispatcher move to new ztest API

### PR DESCRIPTION
**Pull request by hunterhu3000**
Move offload_dispatcher tests to use new ztest API

Signed-off-by: Hu Zhenyu <zhenyu.hu@intel.com>

### CODE REVIEW

**Review by rlubos — DISMISSED** [review]

File: tests/net/socket/offload_dispatcher/src/main.c [hunk 1]
```diff
@@ -436,27 +436,29 @@ static void test_result_reset(void)
...
 
 	test_sock = zsock_socket(AF_INET, SOCK_STREAM, IPPROTO_TLS_1_2);
 	zassert_true(test_sock >= 0, "Failed to create socket");
 	zassert_false(test_socket_ctx[OFFLOAD_1].socket_called,
 		      "Socket should'nt have been dispatched yet");
+	return NULL;
 }
 
-static void test_socket_teardown(void)
+static void test_socket_teardown(void *dummy)
```
Comment (yperess) [inline tests/net/socket/offload_dispatcher/src/main.c:461]: nit: Add `ARG_UNUSED(dummy)`

File: tests/net/socket/offload_dispatcher/src/main.c [hunk 2]
```diff
@@ -842,65 +844,39 @@ static void test_tls_native_iface_native(void)
...
+ZTEST_SUITE(socket_offload_sendmsg_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_getpeername_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_getsockname_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_so_offload_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_so_native_test_suite, NULL, test_socket_setup_udp, NULL, NULL,
+	    test_socket_teardown);
```
Comment (yperess) [inline tests/net/socket/offload_dispatcher/src/main.c:878]: These all have the exact same setup/teardown, why can't we just use 1 suite?

File: tests/net/socket/offload_dispatcher/src/main.c [hunk 3]
```diff
@@ -842,65 +844,39 @@ static void test_tls_native_iface_native(void)
...
+ZTEST_SUITE(socket_offload_tls_offload_test_suite, NULL, test_socket_setup_tls, NULL, NULL,
+	    test_socket_teardown);
+ZTEST_SUITE(socket_offload_tls_native_test_suite, NULL, test_socket_setup_tls, NULL, NULL,
+	    test_socket_teardown);
```
Comment (yperess) [inline tests/net/socket/offload_dispatcher/src/main.c:882]: Same here, these are the same

**Review by yperess — CHANGES_REQUESTED** [review]
There seems to be a lot of unnecessary test suites generated for this test

File: tests/net/socket/offload_dispatcher/src/main.c (hunk 3 above)
Comment (hunterhu3000) [reply tests/net/socket/offload_dispatcher/src/main.c:882]: > Same here, these are the same

The name looks similar
socket_offload_tls_offload_test_suite contains the case of test_tls_native_iface_offloaded
socket_offload_tls_native_test_suite contains the test of test_tls_native_iface_native
They are different

Comment (hunterhu3000) [reply tests/net/socket/offload_dispatcher/src/main.c:882]: In my code setup / teardown will be run before / after each test.
If combine the test cases, setup / teardown will be run only once for each test suite

File: tests/net/socket/offload_dispatcher/src/main.c (hunk 1 above)
Comment (hunterhu3000) [reply tests/net/socket/offload_dispatcher/src/main.c:461]: OK

File: tests/net/socket/offload_dispatcher/src/main.c (hunk 2 above)
Comment (hunterhu3000) [reply tests/net/socket/offload_dispatcher/src/main.c]: only can combine to 3 suite

**Comment by hunterhu3000** [comment]
In this test case CONFIG_ZTEST_SHFFLE=y can be turned on.
Thanks to review.

**Review by rlubos — DISMISSED** [review]

**Review by mniestroj — DISMISSED** [review]

**Review by enjiamai — DISMISSED** [review]

**Review by aaronemassey — APPROVED** [review]

**Review by mniestroj — APPROVED** [review]

**Review by rlubos — APPROVED** [review]

**Review by enjiamai — APPROVED** [review]

**Comment by hunterhu3000** [comment]
@yperess I have modified per your request and reduced the test case number to 3. If it is OK to you, could you please review and +1? So that the patch can be merged into the mainline. Thanks

**Review by yperess — APPROVED** [review]

**Pull request merged**